psydac/api/fem.py

```diff
@@ -634,20 +664,42 @@ def get_space_indices_from_target(self, domain, target):
     def construct_arguments(self):
 
         tests_basis, tests_degrees, spans, pads = construct_test_space_arguments(self.test_basis)
-        n_elements, quads, quads_degree         = construct_quad_grids_arguments(self.grid)
+        n_elements, quads, quads_degree         = construct_quad_grids_arguments(self.grid, use_weights=False)
 
         global_pads   = self.space.vector_space.pads
 
         element_mats, global_mats = self.allocate_matrices()
         self._global_matrices   = [M._data for M in global_mats]
 
         if self.mapping:
-            mapping   = [e._coeffs._data for e in self.mapping._fields]
+            mapping    = [e._coeffs._data for e in self.mapping._fields]
+            space      = self.mapping._fields[0].space
+            map_degree = space.degree
+            map_span   = [q.spans-s for q,s in zip(space.quad_grids, space.vector_space.starts)]
+            map_basis  = [q.basis for q in space.quad_grids]
+            axis       = self.grid.axis
+            ext        = self.grid.ext
+            points     = self.grid.points
+            if axis is not None:
+                nderiv = self.max_nderiv
+                space  = space.spaces[axis]
+                points = points[axis]
+                boundary_basis = basis_ders_on_quad_grid(
+                        space.knots, space.degree, points, nderiv, space.basis)
+
+                map_basis[axis] = map_basis[axis].copy()
+                map_basis[axis][0:1, :, 0:nderiv+1, 0:1] = boundary_basis
+                if ext == 1:
+                    map_span[axis]    = map_span[axis].copy()
+                    map_span[axis][0] = map_span[axis][-1]
             if self.is_rational_mapping:
                 mapping = [*mapping, self.mapping._weights_field._coeffs._data]
```

Comment:
Same comment here: coding `self.mapping._weights_field._coeffs._data` is bad practice

Reply:
fixed